UNR-3965-Spatial-Position-Update-Modification (#2419)
spatialos/UnrealGDK: CHANGELOG.md

**注意**：自虚幻引擎开发套件 v0.8.0 版本起，其日志提供中英文两个版本。每个日志的中文版本都置于英文版本之后。

## [`x.y.z`] - Unreleased

> - Spatial Position Updates are now sent based on a different logic: previously, a position update was sent to spatial if enough time has passed since the Actor's last update (computed using the SpatialUpdateFrequency) AND if the Actor has moved more than the Position Distance Threshold. This change makes the updates being sent if either of the following is true:

Reviewer: These can go in the `features` section

## [`x.y.z`] - Unreleased

### Breaking changes:

> - The logic for sending Spatial position updates has been changed, the `PositionUpdateFrequency` and the `PositionDistanceThreshold` have now been replaced in the GDK settings by 4 new variables. To update your project:

Reviewer 2: Nit - I'd say "condition" instead of "logic" maybe. The important part o a user is that they configure things differently.

>   1. Read the more detailed description of the change, you can find it by searching "Spatial Position Updates are now sent based on a different logic" inside this document.

Reviewer 2: These steps to update your project don't really walk you through what to do. I'd say the minimum set of things to tell users is what they need to do to get back to what they had before.
i.e. set the min thresholds (pending any possible name change) to be the values these old settings currently have. Then set the maximum to be something much bigger.

  2. Find any project-specific usages of the removed variables mentioned above, and replace them according to the new logic.
  NOTE: If your project does not use custom values for the `PositionUpdateFrequency` or `PositionDistanceThreshold`, then, by default, the updates will be sent with the same frequency as before and no action is required.
- Removed the `OnAuthorityLossImminent` Actor event.

### Features:
- Schema bundles are now generated by default (in addition to the schema descriptor). Specifying a Schema bundle location via `-AdditionalSchemaCompilerArguments="--bundle_out=..."` will result in the bundle being written to both the default location (`build/assembly/schema/schema.sb`) _and_ the location specified by the additional arguments.

> - Spatial Position Updates are now sent based on a different logic: previously, a position update was sent to spatial if enough time has passed since the Actor's last update (computed using the SpatialUpdateFrequency) AND if the Actor has moved more than the Position Distance Threshold. This change makes the updates being sent if either of the following is true:

Reviewer 2: There's some nits for grammar and terminology here. Should probably deal with it last though. Note to come back to this.

  1. The time since the last sent position update is greater than or equal to PositionUpdateThresholdMinSeconds AND the distance travelled since the last update is greater than or equal to PositionUpdateThresholdMinDistance.

>   1. The time since tha last sent position update is greater than or equal to PositionUpdateThresholdMaxSeconds AND the Actor has moved any non-null amount.

Reviewer: ```suggestion
  1. The time since tha last sent position update is greater than or equal to PositionUpdateThresholdMaxSeconds AND the Actor has moved a non-zero amount.
```

## [`x.y.z`] - Unreleased

### Breaking changes:

> - The condition for sending Spatial position updates has been changed, the two variables `PositionUpdateFrequency` and `PositionDistanceThreshold` have now been replaced in the GDK settings by 4 new variables. To update your project:

Reviewer 2: nit - I'd say they've been removed but not specify 4 unless you are going to say what they are.
Saying - they are gone and here are the steps to restore functionality - is good.

## [`0.11.0`] - 2020-07-29

### Breaking changes:

> - The logic for sending Spatial position updates has been changed, the `PositionUpdateFrequency` and the `PositionDistanceThreshold` have now been replaced in the GDK settings by 4 new variables. To update your project:

Reviewer: These should be in a new breaking changes section in the unreleased section
